src/app/pages/send/send.component.tsx

```diff
@@ -88,6 +89,20 @@ export class SendPage extends Component<IProps, IState> {
         }
     }
 
+    public async checkAddr(e) {
+        const input = e.target.value;
+        const regex = /[.](zil|eth)$/m;
+        const extension = input.match(regex);
+        const namicorn = new Namicorn();
```

Comment:
I would move Namicorn instantiation in component constructor, so it will be instantiated only once not on every search.